app/src/main/java/com/github/libretube/helpers/PreferenceHelper.kt

```diff
@@ -143,6 +144,16 @@ object PreferenceHelper {
         ).apply()
     }
 
+    fun getSponsorBlockUUID(): String {
+        var uuid = getString(PreferenceKeys.SPONSORBLOCK_UUID, "")
+        if (uuid.isEmpty()) {
+            // generate a new uuid
+            uuid = UUID.randomUUID().toString()
```

Comment:
Can you change the generation to use something like this https://stackoverflow.com/a/20536819

Reply:
Why is it called UUID on the SponsorBlock api docs if it's not a UUID then? For reference, https://en.m.wikipedia.org/wiki/Universally_unique_identifier.

Reply:
It used to be a UUID but it turns out that they are not guarenteed to not be predictable, which means they are not good for authentication

Reply:
> It used to be a UUID but it turns out that they are not guarenteed to not be predictable, which means they are not good for authentication. Will update docs

Actually, this is false. Most UUID implementation (including Java's) are generated with cryptographically secure random number generator.

Reply:
UUIDs can hold at most 128 bits, but sha256 is used to derive the public userID, which can hold 256 bits. So if UUIDs were used for private userIDs, you'd get much less combinations for the private part than the public part, making it easier to guess the private part.
Also in most of the places in the API docs hosted on the wiki, the term "userID" is used instead of submitter UUID. (it probably should be called "private userID" to clearly separate it from the derived public userID)

Reply:
Which length should the userID be of then?